src/services/pagination/LengthAwarePaginator.js

```diff
@@ -0,0 +1,81 @@
+import Paginator from './Paginator';
+import LesgoException from '../../exceptions/LesgoException';
+
+const FILE = 'Services/pagination/LengthAwarePaginator';
+
+export default class LengthAwarePaginator extends Paginator {
+  /**
+   * Constructor
+   *
+   * @param db
+   * @param sql
+   * @param sqlParams
+   * @param perPage
+   * @param currentPage
+   * @param total
+   */
+  constructor(db, sql, sqlParams, perPage = 10, currentPage = 1, total = null) {
+    if (total !== null && typeof total !== 'number') {
```

Comment:
I don't get it, total param is optional however you don't want this to be null? why not explicitly check only for a number type instead?

Reply:
Because there is an method called `countTotalItems()` if total params is `null`.

Reply:
This was the same issue raised and resolved in an earlier call out. Joshua will be doing the necessary corrections